### PR TITLE
fix: rewrite now() to proctime() when plan sink into table's default expr

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

when there exist `now()` in the defualt expression of a column. If the table is sinked into, rewrite the `now()` to `proctime()` because the streaming execution can not handle `now`.
fix https://github.com/risingwavelabs/risingwave/issues/17278

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by BugenZhao — COMMENTED** [review]
Interesting.

Is it because we need to evaluate `NOW` for the records from upstream sinks, compared to DML which is evaluated in batch executors?

If so, then I'm wondering how or whether we've handled `DEFAULT NOW()` in tables with connectors...

**Comment by st1page** [comment]
> Interesting.
> 
> Is it because we need to evaluate `NOW` for the records from upstream sinks, compared to DML which is evaluated in batch executors?
> 
> If so, then I'm wondering how or whether we've handled `DEFAULT NOW()` in tables with connectors...

Actually I am not sure how we have handled default column in tables with connector 🤔

File: src/frontend/planner_test/tests/testdata/output/sink_into_table.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# This file is automatically generated. See `src/frontend/planner_test/README.md` for more information.
+- id: create sink into table having default expression with now()
```
Comment (BugenZhao) [inline src/frontend/planner_test/tests/testdata/output/sink_into_table.yaml:2]: Do we have existing tests for default values in `SINK INTO TABLE` besides `NOW()`?

**Review by BugenZhao — APPROVED** [review]

File: src/frontend/planner_test/tests/testdata/output/sink_into_table.yaml (hunk 1 above)
Comment (st1page) [reply src/frontend/planner_test/tests/testdata/output/sink_into_table.yaml:2]: In e2e_test/sink/sink_into_table we test default values and generated columns. But we do not have many planner test, I will take a look to add some.

**Pull request merged**